modules/API/models/CardDAV.php

```diff
@@ -277,48 +319,64 @@ public function deletedCard($card)
 		\App\Log::trace(__METHOD__ . ' | End');
 	}
 
-	public function createRecord($moduleName, $card)
+	/**
+	 * Create record.
+	 *
+	 * @param string $moduleName
+	 * @param array  $card
+	 *
+	 * @return void
+	 */
+	public function createRecord(string $moduleName, array $card)
 	{
 		\App\Log::trace(__METHOD__ . ' | Start Card ID' . $card['id']);
 		$vcard = Sabre\VObject\Reader::read($card['carddata']);
-		$leadId = false;
-		if (isset($vcard->ORG)) {
+		$record = Vtiger_Record_Model::getCleanInstance($moduleName);
+		if ($moduleName==='Contacts' && isset($vcard->ORG)) {
 			$lead = Vtiger_Record_Model::getCleanInstance('Leads');
 			$lead->set('assigned_user_id', $this->user->get('id'));
-			$lead->set('company', \App\Purifier::purify((string) $vcard->ORG));
-			$lead->set('lastname', \App\Purifier::purify((string) $vcard->ORG));
 			$lead->set('leadstatus', 'PLL_PENDING');
 			$lead->set('vat_id', '');
+			$lead->setDBValue('company', \App\Purifier::purify((string) $vcard->ORG));
 			$lead->save();
-			$leadId = $lead->getId();
+			$fieldModel = current($record->getModule()->getReferenceFieldsForModule('Leads'));
+			if ($fieldModel) {
+				$record->set($fieldModel->getFieldName(), $lead->getId());
+			}
 		}
-		$head = $vcard->N->getParts();
-
-		$record = Vtiger_Record_Model::getCleanInstance($moduleName);
 		$record->set('assigned_user_id', $this->user->get('id'));
-		if ('Contacts' === $moduleName) {
-			$record->set('firstname', \App\Purifier::purify($head[1]));
-			$record->set('lastname', \App\Purifier::purify($head[0]));
-			$record->set('jobtitle', \App\Purifier::purify((string) $vcard->TITLE));
-		} elseif ('OSSEmployees' === $moduleName) {
-			$record->set('name', \App\Purifier::purify($head[1]));
-			$record->set('last_name', \App\Purifier::purify($head[0]));
-		}
-		$record->set('description', \App\Purifier::purify((string) $vcard->NOTE));
-		if ($leadId) {
-			$record->set('parent_id', $leadId);
-		}
-		foreach ($this->telFields[$moduleName] as $key => $val) {
-			$record->set($key, $this->getCardTel($vcard, $val));
-		}
-		foreach ($this->mailFields[$moduleName] as $key => $val) {
-			$record->set($key, $this->getCardMail($vcard, $val));
-		}
-		if (isset($vcard->ADR)) {
-			$this->setRecordAddres($vcard, $moduleName, $record);
-		}
+		$this->setValuesForRecord($record, $vcard);
 		$record->save();
+		$this->updateIdAndModifiedTime($record, $card);
+		\App\Log::trace(__METHOD__ . ' | End');
+	}
 
+	/**
+	 * Update record.
+	 *
+	 * @param Vtiger_Record_Model $record
+	 * @param array               $card
+	 */
+	public function updateRecord(Vtiger_Record_Model $record, array $card)
+	{
+		\App\Log::trace(__METHOD__ . ' | Start Card ID:' . $card['id']);
+		$vcard = Sabre\VObject\Reader::read($card['carddata']);
```

Comment:
move to App\Integrations\Dav\Card::loadFromContent
similar to App\Integrations\Dav\Calendar::loadFromContent